### PR TITLE
Migrate Old (UserProfile.name_en) into (UserProfile.meta)

### PR DESCRIPTION
**Pull request by shadinaif**
Add new command to copy database old data of **UserProfile.name_en** into **UserProfile.meta**, and then drop the old column (**name_en**)

This can be called by the following command:
```
python manage.py lms edraak_migrate_userprofile_name_en
```

------------
Related to https://edraak.atlassian.net/browse/OU-346

### CODE REVIEW

**Review by ghost — CHANGES_REQUESTED** [review]
Thanks @shadinaif! Please convert to a [Django management command](https://docs.djangoproject.com/en/3.0/howto/custom-management-commands/) instead. Here's why:

 - Add migrations on apps that we don't own (i.e. edX Apps like `students`) would mean that every time upstream makes a new migration we need to [create yet another "merge" migration](https://riptutorial.com/django/example/23659/solving-migration-conflicts). Merge migrations are very painful to deal with and I recommend avoiding them unless absolutely necessary.

 - The reason why we chose to override `meta` is to avoid having a new migration like what this PR is introducing.
 - Unlike migrations, Django commands do no run automatically, which is an acceptable compromise in my opinion.

Please let me know if you have any questions.

**Comment by shadinaif** [comment]
All done @OmarEdraak . Please review again

File: common/djangoapps/student/management/commands/edraak_migrate_userprofile_name_en.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            old_data = UserProfile.objects.raw(
+                "SELECT id, meta, name_en FROM auth_userprofile WHERE (name_en IS NOT NULL) AND (name_en != '');"
+            )
+
+            # Copying from (name_en) to (meta)
+            for user_profile in old_data:
+                meta = json.loads(user_profile.meta or '{}')
+                meta['edraak_name_en'] = user_profile.name_en
+                user_profile.meta = json.dumps(meta)
+                user_profile.save()
```
Comment (ghost) [inline common/djangoapps/student/management/commands/edraak_migrate_userprofile_name_en.py]:
```suggestion
            for user_profile in old_data:
                user_profile.name_en = user_profile.DB_name_en
                user_profile.save()
```

File: common/djangoapps/student/management/commands/edraak_migrate_userprofile_name_en.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+class Command(BaseCommand):
+    help = 'Migrate database column (auth_userprofile.name_en) into (auth_userprofile.meta) and drop the column'
+
+    def handle(self, *args, **options):
+        if check_name_en():
+            print('Database column (name_en) found in (auth_userprofile) table. Applying migration...')
+
+            old_data = UserProfile.objects.raw(
+                "SELECT id, meta, name_en FROM auth_userprofile WHERE (name_en IS NOT NULL) AND (name_en != '');"
```
Comment (ghost) [inline common/djangoapps/student/management/commands/edraak_migrate_userprofile_name_en.py]:
```suggestion
                "SELECT id, meta, name_en as DB_name_en FROM auth_userprofile WHERE (name_en IS NOT NULL) AND (name_en != '');"
```

**Review by ghost — APPROVED** [review]
`name_en` conflicts with the existing `@property def name_en`, so I used SQL aliases to solve the issue.

This saves us the effort to manually working with meta and use the `@name_en.setter` property instead.

See:

https://github.com/Edraak/edraak-platform/blob/db4d256e29bb204defff301d8ca394a8e62c5fc6/common/djangoapps/student/models.py#L465-L485

File: common/djangoapps/student/management/commands/edraak_migrate_userprofile_name_en.py (hunk 1 above)
Comment (shadinaif) [reply common/djangoapps/student/management/commands/edraak_migrate_userprofile_name_en.py]: Done

File: common/djangoapps/student/management/commands/edraak_migrate_userprofile_name_en.py (hunk 2 above)
Comment (shadinaif) [reply common/djangoapps/student/management/commands/edraak_migrate_userprofile_name_en.py]: Done

**Review by ghost — APPROVED** [review]

**Pull request merged**